documentation/Arbitraries.md

```diff
@@ -3075,6 +3075,26 @@ fc.letrec(tie => ({
 // • {"left":{"left":1517743480,"right":1007350543},"right":{"left":12,"right":2147483643}}
 // • …
 ```
+
+```ts
+import * as fc from 'fast-check';
+
+// Specify recursive type explicitly, since TS cannot infer recursive types
+type Tree = number | { left: Tree; right: Tree };
+// Explicitly specify types of each recursive constructors
+const { tree } = fc.letrec<{
+  tree: Tree;
+  node: { left: Tree; right: Tree };
+  leaf: number;
+}>((tie) => ({
+  tree: fc.oneof(tie('node'), tie('leaf'), tie('leaf')),
+  node: fc.record({
+    left: tie('tree'),
+    right: tie('tree'),
+  }),
+  leaf: fc.nat(),
+}));
+```
```

Comment:
Instead of the example in the doc Arbitrary.md mostly explaining the primitives from a JS point of view, I'd rather update the example in the `@example` of `letrec` to add the typing trick.

Something like:
```txt
 * // Specify recursive type explicitly, since TS cannot infer recursive types
 * type Tree = number | [Tree, Tree];
 * // Explicitly specify types of each recursive constructors
 * const { tree } = fc.letrec<{
 *   tree: Tree;
 *   node: [Tree, Tree];
 *   leaf: number;
 * }>(tie => ({
 *   tree: fc.oneof({depthFactor: 0.5}, tie('leaf'), tie('node')),
 *   node: fc.tuple(tie('tree'), tie('tree')),
 *   leaf: fc.nat()
 * }));
 * // tree is 50% of node, 50% of leaf
 * // the ratio goes in favor of leaves as we go deeper in the tree (thanks to depthFactor)
```

As the pattern `fc.name<ExpectedType>` is mode and more spreading, having it only inside the jsdoc looks legit.